From: Jakub Jirutka <jakub@jirutka.cz>
Date: Tue, 6 Apr 2021 00:24:00 +0200
Subject: [PATCH] Reduce size of the binary

Reduce size of the resulting binary: 4.6 -> 3.0 MiB (on x86_64).

`opt-level="s"` provides the best size/performance ratio; the performance
is close to the default opt-level (according to the benchmark), but the
binary is significantly smaller.

NOTE: ripgrep uses jemalloc allocator when building on musl libc.
It adds about 200 kiB to the binary size, but provides 2x performance;
so better keep it as-is.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -78,7 +78,10 @@
 pcre2 = ["grep/pcre2"]
 
 [profile.release]
-debug = 1
+lto = true
+codegen-units = 1
+panic = "abort"
+opt-level = "s"
 
 [package.metadata.deb]
 features = ["pcre2"]
